pkg/uconnproxy/mongodb.go

```diff
@@ -0,0 +1,96 @@
+package uconnproxy
+
+import (
+	"runtime"
+
+	"github.com/activecm/rita/resources"
+	"github.com/activecm/rita/util"
+	"github.com/globalsign/mgo"
+	"github.com/vbauerster/mpb"
+	"github.com/vbauerster/mpb/decor"
+)
+
+type repo struct {
+	res *resources.Resources
+}
+
+// NewMongoRepository create new repository
+func NewMongoRepository(res *resources.Resources) Repository {
+	return &repo{
+		res: res,
+	}
+}
+
+func (r *repo) CreateIndexes() error {
+	session := r.res.DB.Session.Copy()
+	defer session.Close()
+
+	// set collection name
+	collectionName := r.res.Config.T.Structure.UniqueConnProxyTable
+
+	// check if collection already exists
+	names, _ := session.DB(r.res.DB.GetSelectedDB()).CollectionNames()
+
+	// if collection exists, we don't need to do anything else
+	for _, name := range names {
+		if name == collectionName {
+			return nil
+		}
+	}
+
+	indexes := []mgo.Index{
+		{Key: []string{"src", "src_network_uuid", "fqdn"}, Unique: true},
```

Comment:
Please  swap the ordering of this key to be 
`{Key: []string{"src", "fqdn", "src_network_uuid"}, Unique: true},`

MongoDB can use the prefix of a compound key to service a query. By reordering this, we gain the ability to search for (src, fqdn) pairs without the network uuid which can be helpful for quick debugging. In the current ordering, we have the ability to search for (src, src_network_uuid), but that search is already covered by the key below.

Reply:
I was not aware of that, good to know! Re-ordered